src/sentry/ingest/transaction_clusterer/datasource.py

```diff
@@ -0,0 +1,36 @@
+from datetime import datetime
+from typing import Iterable, Tuple
+
+from snuba_sdk import Column, Condition, Entity, Limit, Op, Query, Request
+
+from sentry.models import Project
+from sentry.utils.snuba import raw_snql_query
+
+TRANSACTION_SOURCE = "url"
+
+
+def fetch_unique_transaction_names(
+    project: Project, time_range: Tuple[datetime, datetime], limit: int
+) -> Iterable[str]:
+    then, now = time_range
+    snuba_request = Request(
+        "transactions",
+        app_id="transactions",
+        query=Query(
+            match=Entity("transactions"),
+            select=[Column("transaction")],
+            where=[
+                Condition(Column("project_id"), Op.EQ, project.id),
+                Condition(Column("finish_ts"), Op.GTE, then),
+                Condition(Column("finish_ts"), Op.LT, now),
+                Condition(Column("transaction_source"), Op.EQ, TRANSACTION_SOURCE),
+            ],
+            groupby=[Column("transaction")],
+            limit=Limit(limit),
+        ),
+    )
+    snuba_response = raw_snql_query(
+        snuba_request, referrer="src.sentry.ingest.transaction_clusterer"
```

Comment:
Should the referrer have any additional information to identify the datasource?

Reply:
Referrers should be string literals AFAIK: https://github.com/getsentry/sentry/blob/439f1beced91f46c5d0f1e2e7192d0a538067ed6/src/sentry/release_health/metrics.py#L370-L372